forc-plugins/forc-node/src/ignition/op.rs

```diff
@@ -0,0 +1,106 @@
+use super::cmd::IgnitionCmd;
+use crate::{
+    chain_config::{check_and_update_chain_config, ChainConfig},
+    consts::{
+        MAINNET_BOOTSTRAP_NODE, MAINNET_RELAYER_DA_DEPLOY_HEIGHT,
+        MAINNET_RELAYER_LISTENING_CONTRACT, MAINNET_RELAYER_LOG_PAGE_SIZE, MAINNET_SERVICE_NAME,
+        MAINNET_SYNC_HEADER_BATCH_SIZE,
+    },
+    run_opts::{DbType, RunOpts},
+    util::{ask_user_keypair, ask_user_string, HumanReadableCommand, KeyPair},
+};
+use anyhow::Context;
+use forc_tracing::println_green;
+use std::{
+    net::IpAddr,
+    path::PathBuf,
+    process::{Child, Command},
+};
+/// Configures the node with testnet configuration to connect the node to latest testnet.
+/// Returns `None` if this is a dry_run and no child process created for fuel-core.
+pub(crate) async fn run(cmd: IgnitionCmd, dry_run: bool) -> anyhow::Result<Option<Child>> {
+    check_and_update_chain_config(ChainConfig::Testnet).await?;
+    let keypair = if let (Some(peer_id), Some(secret)) = (
+        &cmd.connection_settings.peer_id,
+        &cmd.connection_settings.secret,
+    ) {
+        KeyPair {
+            peer_id: peer_id.to_string(),
+            secret: secret.to_string(),
+        }
+    } else {
+        ask_user_keypair()?
+    };
+
+    let relayer = if let Some(relayer) = cmd.connection_settings.relayer {
+        relayer
+    } else {
+        ask_user_string("Ethereum RPC (mainnet) Endpoint:")?
+    };
```

Comment:
nit:
```suggestion
    let relayer = cmd.connection_settings.relayer.unwrap_or_else(|| ask_user_string("Ethereum RPC (mainnet) Endpoint:").unwrap());
```

Reply:
There is a semantic difference betweent he original piece and suggestion. 

Suggestion panics. Original one propagetes the error. If you want to ask for a single line/functional style matching something like:

```rust
let relayer = cmd.connection_settings.relayer
    .map_or_else(
        || ask_user_string("Ethereum RPC (mainnet) Endpoint:"),
        Ok
    )?;
```
is needed because you need to be returning a result from a closure, and unfortunately rust is not very good at this as it has a rigid type system and no implicit return type coercion.

Reply:
we can preserve error like so i think - but i havent followed through whether it is the same error type:
```rust
let relayer = cmd.connection_settings.relayer.unwrap_or_else(|| ask_user_string("Ethereum RPC (mainnet) Endpoint:"))?;
```
again; a nit - not required.